### PR TITLE
kraken: Fix second pass overfilter

### PR DESCRIPTION
**Pull request by azime**
http://jira.canaltp.fr/browse/NAVITIAII-2062

When multiple arrival with [same walking, same time, same number of sections] were possible we used to keep only the first discovered, but it's better (and not expensive) to keep them all, as they are all equally interesting.
We also avoid stay_in problems (see JIRA)

### CODE REVIEW

File: source/routing/tests/routing_api_test.cpp [hunk 1]
```diff
@@ -2187,3 +2187,106 @@ BOOST_FIXTURE_TEST_CASE(direct_path_car, streetnetworkmode_fixture<test_speed_pr
     BOOST_REQUIRE_EQUAL(resp.journeys(0).sections_size(), 5);
 
 }
+/*
```
Comment (antoine-de) [inline source/routing/tests/routing_api_test.cpp]: great art! :+1:

File: source/routing/tests/routing_api_test.cpp [hunk 2]
```diff
@@ -2187,3 +2187,106 @@ BOOST_FIXTURE_TEST_CASE(direct_path_car, streetnetworkmode_fixture<test_speed_pr
...
+                                                                |          ||              |
+                                                                |          ||              |
+                                                                |    StopPoint:B1          |
+                                                                |          ||              |
+                                                                |--------------------------|
+                                                                           ||
+                                                                           ||
+                                                                           ||
+  |-------------|                                                          ||
+  | StopArea:A  |                         vj:B                             ||
```
Comment (antoine-de) [inline source/routing/tests/routing_api_test.cpp]: StopArea:C no ?

Comment (azime) [inline source/routing/tests/routing_api_test.cpp]: Yes !

**Comment by azime** [comment]
corrections

File: source/routing/tests/raptor_test.cpp [hunk 3]
```diff
@@ -2793,3 +2793,98 @@ BOOST_AUTO_TEST_CASE(reader_with_invalid_vj_extensions) {
...
+    b.data->pt_data->index();
+    b.finish();
+    b.data->build_raptor();
+    b.data->build_uri();
+    RAPTOR raptor(*(b.data));
+    const type::PT_Data& d = *b.data->pt_data;
+
+    routing::map_stop_point_duration departures, arrivals;
+
+    departures[SpIdx(*d.stop_areas_map.at("stop_area:A")->stop_point_list.front())] = 0_min;
```
Comment (antoine-de) [inline source/routing/tests/raptor_test.cpp]: minor: you can use the new builder's getter:

```
departures[SpIdx(*b.get<nt::StopPoint>("stop_point:A")] = 0_min
arrivals[SpIdx(*b.get<nt::StopPoint>("stop_point:B1")] = 0_min
arrivals[SpIdx(*b.get<nt::StopPoint>("stop_point:B2")] = 0_min
```

File: source/routing/tests/raptor_test.cpp [hunk 4]
```diff
@@ -2793,3 +2793,98 @@ BOOST_AUTO_TEST_CASE(reader_with_invalid_vj_extensions) {
...
+BOOST_AUTO_TEST_CASE(fix_datetime_represents_arrival_departure) {
+    ed::builder b("20120614");
+
+    b.sa("stop_area:A", 0, 0, false)("stop_point:A");
+    b.sa("stop_area:B", 0, 0, false)("stop_point:B1")("stop_point:B0");
+    b.sa("stop_area:C", 0, 0, false)("stop_point:C");
+
+    b.vj("A", "1111111", "block1", true, "vj:A")("stop_point:A", "8:05"_t, "8:05"_t)
+            ("stop_point:B0", "08:40"_t, "08:40"_t);
+    b.vj("A", "1111111", "block1", true, "vj:B")("stop_point:B1", "08:40"_t, "09:05"_t)
```
Comment (antoine-de) [inline source/routing/tests/raptor_test.cpp]: minor:

you can use the new composite setter:

`b.vj("A").name("vj:A").block("block1")("stop_point:A", "8:05"_t, "8:05"_t)("stop_point:B0", "08:40"_t, "08:40"_t);`

(but I think if you do so, you'll need to add the `block` setter)

Comment (azime) [inline source/routing/tests/raptor_test.cpp]: OK

File: source/routing/tests/raptor_test.cpp (hunk 3 above)
Comment (azime) [inline source/routing/tests/raptor_test.cpp]: OK

**Comment by azime** [comment]
corrections

File: source/routing/raptor.cpp [hunk 5]
```diff
@@ -212,6 +212,11 @@ struct Dom {
     bool clockwise;
     typedef std::pair<size_t, StartingPointSndPhase> Arg;
     inline bool operator()(const Arg& lhs, const Arg& rhs) const {
+        if (lhs.second.count == rhs.second.count
```
Comment (pbougue) [inline source/routing/raptor.cpp:219]: A comment might be nice to tell why we do that (you can get inspiration in the comment of that PR)

File: source/routing/tests/raptor_test.cpp [hunk 6]
```diff
@@ -2793,3 +2793,97 @@ BOOST_AUTO_TEST_CASE(reader_with_invalid_vj_extensions) {
...
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().arrival, time_from_string("2012-06-14 08:40:00"));
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().stop_points.front()->uri, "stop_point:A");
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().stop_points.back()->uri, "stop_point:B0");
+
+    BOOST_REQUIRE_EQUAL(resp_0.size(), resp_1.size());
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.size(), resp_1.at(0).items.size());
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().departure, resp_1.at(0).items.front().departure);
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().arrival, resp_1.at(0).items.front().arrival);
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().type, resp_1.at(0).items.front().type);
+    BOOST_CHECK_EQUAL(resp_0.at(0).items.front().stop_points.front()->uri, resp_1.at(0).items.front().stop_points.front()->uri);
```
Comment (pbougue) [inline source/routing/tests/raptor_test.cpp]: minor: line length

**Comment by antoine-de** [comment]
OK for me

**Comment by azime** [comment]
comment added

**Pull request merged**